tests/test_admin.rs

```diff
@@ -347,6 +347,263 @@ async fn test_topics() {
+
+        let res = admin_client
+            .create_topics(&[topic1, topic2], &opts)
+            .await
+            .expect("topic creation failed");
+
+        assert_eq!(res, &[Ok(name1.clone()), Ok(name2.clone())]);
+        let metadata1 = fetch_metadata(&name1);
+        let metadata2 = fetch_metadata(&name2);
+        assert_eq!(1, metadata1.topics().len());
+        assert_eq!(1, metadata2.topics().len());
+        let metadata_topic1 = &metadata1.topics()[0];
+        let metadata_topic2 = &metadata2.topics()[0];
+        assert_eq!(&name1, metadata_topic1.name());
+        assert_eq!(&name2, metadata_topic2.name());
+
+        populate_topic(
+            &name1,
+            message_count,
+            &value_fn,
+            &key_fn,
+            Some(partition),
+            None,
+        )
+        .await;
+        populate_topic(
+            &name2,
+            message_count,
+            &value_fn,
+            &key_fn,
+            Some(partition),
+            None,
+        )
+        .await;
+
+        let mut offsets = TopicPartitionList::new();
+        offsets
+            .add_partition_offset(&name1, partition, Offset::Offset(message_count.into()))
+            .expect("add partition offset1 failed");
+        let unknown_partition = 42;
+        offsets
+            .add_partition_offset(
+                &name2,
+                unknown_partition,
+                Offset::Offset(message_count.into()),
+            )
+            .expect("add partition offset2 failed");
+        let results = admin_client
+            .delete_records(&offsets, &opts)
+            .await
+            .expect("delete records failed");
+
+        // Results can be in any order that is why we can't just say the following:
+        // assert_eq!(res, &[Ok(name1.clone()), Err((name2.clone(), RDKafkaErrorCode::UnknownPartition))]);
+
+        let mut found = false;
+        for result in &results {
+            if let Ok(name) = result {
+                if *name == name1 {
+                    found = true;
+                    break;
+                }
+            }
+        }
+        assert!(found);
+
+        let mut found = false;
+        for result in &results {
+            if let Err((name, err)) = result {
+                if *name == name2 {
+                    assert_eq!(err, &RDKafkaErrorCode::UnknownPartition);
+                    found = true;
+                    break;
+                }
+            }
+        }
+        assert!(found);
+    }
+
+    // Verify that deleting records from a non-existent topic fails.
+    {
+        let name = rand_test_topic();
+        let mut offsets = TopicPartitionList::new();
+        offsets
+            .add_partition_offset(&name, 0, Offset::Offset(0))
+            .expect("add partition offset failed");
+        let res = admin_client.delete_records(&offsets, &opts).await;
+        assert_eq!(res, Err(KafkaError::AdminOp(RDKafkaErrorCode::NoEnt)));
+    }
+}
```

Comment:
This is too much code to duplicate to test this feature. Could you instead work the `delete_records` bit into `test_topics`?

Reply:
@benesch Updated tests.

Reply:
@benesch any update on the review?

Reply:
Hi @benesch is there anything I can do to help merge this?